drivers/sqlite3/sqshared/reader.go

```diff
@@ -125,8 +126,18 @@ FROM (
 		conds = append(conds, "table_name LIKE ?")
 	}
 	if len(f.Types) != 0 {
+		tableTypes := []string{
+			"BASE TABLE",
+			"TABLE",
+			"VIEW",
+			"GLOBAL TEMPORARY",
+		}
 		pholders := []string{}
 		for _, t := range f.Types {
+			if !slices.Contains(tableTypes, t) {
```

Comment:
This prevents from getting system tables at all. We should check `f.WithSystem` here.